akka-http-core/src/main/scala/akka/http/javadsl/ConnectHttp.scala

```diff
@@ -167,6 +168,9 @@ final class ConnectHttpImpl(val host: String, val port: Int, val http2: UseHttp2
   def isHttps: Boolean = false
 
   def connectionContext: Optional[HttpsConnectionContext] = Optional.empty()
+
+  override def effectiveConnectionContext(fallbackContext: ConnectionContext): ConnectionContext =
+    HttpConnectionContext(http2.asScala)
```

Comment:
So this always ignores the `host`, `port`, `connectionContext` and the `fallbackContext` parameter? That seems odd..

Reply:
ah `connectionContext` is the HTTPS context so that might make sense to ignore here.

Reply:
:spades: 

This did take me a couple of hours :confused: 

I find the API very confusing bewtween the ConnectHttp and Context.

the `connectionContext` is only Https so a user can't ever have called `withContext` and set a custom one here.

I think that is a restriction in the API that we could remove but it wouldn't be binary/source compatible so this was a quick way to make it work without breaking any existing users.

Reply:
The host and port are ignored here but when the delegation to the scaladsl from the javadsl they are taken from the ConnectHttp and passed in

Reply:
Yeah seems to make sense, just trying to digest the change ;)